File: Back-End/BookStoreApp.Core/Entity/Book.cs
using BookStoreApp.Core.ValueObjects;
using System;

namespace BookStoreApp.Core.Entity
{
    public class Book : BaseEntity
    {
        public string Name { get; private set; }
        public Guid AuthorId { get; private set; }
        public Category Category { get; private set; }
        public string Language { get; private set; } 
        public PublicationDate PublicationDate { get; private set; } 
        public string CoverPicture { get; private set; } //Change to appropriate Data type byte[]/string/File etc..
        
        public Book(Guid id,string name, Guid authorId, Category category, string language, PublicationDate publicationDate, string coverPicture)
        {
            //Guards & Exceptions
            Id = id;
            Name = name;
            AuthorId = authorId;
            Category = category;
            Language = language;
            PublicationDate = publicationDate;
            CoverPicture = coverPicture;
        }

        //Update book picture.
        //Update book.
    }
}

File: Front-End/src/app/Models/Book/Book.ts
import { Guid } from 'guid-typescript';
import { Category } from '../Category';
import { PublicationDate } from '../PublicationDate';

export interface Book {
  id: Guid;
  name: string;
  authorId: Guid;
  category: Category;
  language: string;
  about: string;
  publicationDate: PublicationDate;
  coverPicture: string;
}
